src/utils.cc

```diff
@@ -564,6 +564,7 @@ TemporaryFile::~TemporaryFile() { boost::filesystem::remove(tmp_name_); }
 
 void TemporaryFile::PutContents(const std::string &contents) {
   std::ofstream out(Path().c_str());
+  chmod(tmp_name_.c_str(), S_IRUSR | S_IWUSR);
```

Comment:
We shouldn't `chmod` a file after we create it. There could be a race between creation and access where someone can do nasty things.